File: llm-gateway/src/main/java/com/llmgateway/llm/dto/ask/AskRequestDto.java
package com.llmgateway.llm.dto.ask;

import jakarta.validation.constraints.NotBlank;

public record AskRequestDto (
    @NotBlank(message = "Language can't be blank")
    String language,
    @NotBlank(message = "Code can't be blank")
    String code,
    @NotBlank(message = "Question can't be blank")
    String question
){}

File: backend/src/modules/llm/dto/ask-request.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString } from 'class-validator';

export class AskRequestDto {
  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  language!: string;

  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  code!: string;

  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  message!: string;
}
